sinks/influxdb/driver.go

```diff
@@ -85,101 +94,83 @@ func (self *influxdbSink) metricToSeries(timeseries *sink_api.Timeseries) *influ
 	if timeseries.MetricDescriptor.Type.String() != "" {
 		seriesName = fmt.Sprintf("%s_%s", seriesName, timeseries.MetricDescriptor.Type.String())
 	}
-
-	// Add the real metric value.
-	columns = append(columns, "value")
-	values = append(values, timeseries.Point.Value)
+	point := influxdb.Point{
+		Measurement: seriesName,
+		Tags:        make(map[string]string, 0, len(timeseries.Point.Labels)),
+		Fields: map[string]interface{}{
+			"value": timeseries.Point.Value,
+		},
+		Time: timeseries.Point.End.UTC(),
+	}
 	// Append labels.
-	if !self.c.avoidColumns {
-		for key, value := range timeseries.Point.Labels {
-			columns = append(columns, key)
-			values = append(values, value)
+	for key, value := range timeseries.Point.Labels {
+		if value != "" {
+			point.Tags[key] = value
 		}
-	} else {
-		seriesName = strings.Replace(seriesName, "/", "_", -1)
-		seriesName = fmt.Sprintf("%s_%s", util.LabelsToString(timeseries.Point.Labels, "_"), seriesName)
 	}
-	// Add timestamp.
-	columns = append(columns, "time")
-	values = append(values, timeseries.Point.End.Unix())
-	// Ass sequence number
-	columns = append(columns, "sequence_number")
-	values = append(values, self.seqNum.Get(seriesName))
 
-	return self.newSeries(seriesName, columns, values)
-}
-
-var eventColumns = []string{
-	"time",                     // Column 0
-	"sequence_number",          // Column 1
-	sink_api.LabelPodId.Key,    // Column 2
-	sink_api.LabelPodName.Key,  // Column 3
-	sink_api.LabelHostname.Key, // Column 4
-	"value",                    // Column 5
+	return point
 }
 
 // Stores events into the backend.
 func (sink *influxdbSink) StoreEvents(events []kube_api.Event) error {
-	dataPoints := []*influxdb.Series{}
+	if err := sink.createDatabase(); err != nil {
```

Comment:
Registration is done synchronously as part of startup. Even if we
successfully create db as part of startup, what if influxDB crashed and
came back up? The previously created DB would no longer exist. These
changes make heapster resilient to DB failures.

On Wed, Sep 16, 2015 at 3:19 PM, Daniel Martí <notifications@github.com>
wrote:

> In sinks/influxdb/driver.go
> <https://github.com/kubernetes/heapster/pull/548#discussion_r39692705>:
>
> >  }
> >
> >  // Stores events into the backend.
> >  func (sink *influxdbSink) StoreEvents(events []kube_api.Event) error {
> > -	dataPoints := []*influxdb.Series{}
> > +	if err := sink.createDatabase(); err != nil {
>
> I would expect heapster to refuse to use a sink if its registration fails,
> or to retry a few times.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/heapster/pull/548/files#r39692705>.
>